modules/diversion_qr_to_waypoint_list.py

```diff
@@ -0,0 +1,49 @@
+"""
+Function to parse diversion QR string into a list of vertices and a rejoin waypoint.
+"""
+import re
+
+
+def diversion_qr_to_waypoint_list(qr_text: str) -> "tuple[list[Waypoint], Waypoint] | None":

```

Comment:
Switch to the `result, data` pattern, return: `"tuple[bool, tuple[list[str], str] | None"`

If it wasn't able to parse, the return would be `False, None`

If it was able to parse, the return would be `True, (["waypoint 1", "name"], "Rejoin at this waypoint")`